spec/core_ext/object/descendants_spec.rb

```diff
@@ -0,0 +1,20 @@
+describe Class do
+  context ".descendant_get" do
+    before do
+      class Yy; end
```

Comment:
I guess my concern is that too many times we have run into constant collision in tests and it is hard to debug.  If you do something like the following, it can be avoided.

```ruby
let!(:base_class) do
  Class.new do
    def self.name
      "Base"
    end
  end
end
let!(:subclass_class) do
  Class.new(base_class) do
    def self.name
      "SubClass"
    end
  end
end


it "returns the correct descendant" do
  expect(base_class.descendant_get("SubClass")).to eq(subclass_class)
end
```